src/app/DIAmeterApplication.cpp

```diff
@@ -73,241 +80,871 @@ int DIAmeterApplication::main(const vector<string>& input_files, const string in
+   /* vector<pair<double*, double*>> intensity_arrs_vector;
+   for (pair<vector<int>::const_iterator, vector<int>::const_iterator> f(valid_ms1scans.begin(), valid_ms2scans.begin());
+       f.first != valid_ms1scans.end() && f.second != valid_ms2scans.end(); ++f.first, ++f.second) {
+
+	   int curr_ms1scan = *(f.first);
+	   int curr_ms2scan = *(f.second);
+	   // carp(CARP_DETAILED_DEBUG, "curr_ms1scan:%d \t curr_ms2scan:%d", curr_ms1scan, curr_ms2scan);
+
+	   // calculate precursor intensity logrank
+	   double *ms1_intensity_arr = NULL;
+	   map<int, pair<double*, double*>>::iterator ms1_intensityIter = ms1scan_intensity_rank_map->find(curr_ms1scan);
+	   if (ms1_intensityIter == ms1scan_intensity_rank_map->end()) { carp(CARP_DETAILED_DEBUG, "No intensity found in MS1 scan:%d !!!", curr_ms1scan); }
+	   else { ms1_intensity_arr = (ms1_intensityIter->second).first; }
+
+	   double *ms2_intensity_arr = NULL;
+	   map<int, double*>::iterator ms2_intensityIter = ms2scan_intensity_map->find(curr_ms2scan);
+	   if (ms2_intensityIter == ms2scan_intensity_map->end()) { carp(CARP_DETAILED_DEBUG, "No intensity found in MS2 scan:%d !!!", curr_ms2scan); }
+	   else { ms2_intensity_arr = ms2_intensityIter->second; }
+
+	   if (ms1_intensity_arr != NULL and ms2_intensity_arr != NULL) {
+		   intensity_arrs_vector.push_back(make_pair(ms1_intensity_arr, ms2_intensity_arr));
+		   // carp(CARP_DETAILED_DEBUG, "max_ms1_intensity:%f \t max_ms2_intensity:%f", MathUtil::MaxInArr(ms1_intensity_arr, max_ms1_mzbin_), MathUtil::MaxInArr(ms2_intensity_arr, max_ms2_mzbin_));
+	   }
+
+   }
+   map<TideMatchSet::Arr::iterator, boost::tuple<double, double, double>> coelute_map;
+   computePrecFragCoeluteOld(targets, peptides, &intensity_arrs_vector, &coelute_map, charge);
+   computePrecFragCoeluteOld(decoys, peptides, &intensity_arrs_vector, &coelute_map, charge); */
+
+   // Loop through each corresponding ms1scan and ms2scan pair (New version which is based on ppm)
+   vector<boost::tuple<double*, double*, int, double*, double*, int>> mz_intensity_arrs_vector;
+   for (pair<vector<int>::const_iterator, vector<int>::const_iterator> f(valid_ms1scans.begin(), valid_ms2scans.begin());
+       f.first != valid_ms1scans.end() && f.second != valid_ms2scans.end(); ++f.first, ++f.second) {
+
+	   int curr_ms1scan = *(f.first);
+	   int curr_ms2scan = *(f.second);
+
+	   int ms1_peak_num = -1; double *ms1_mz_arr = NULL, *ms1_intensity_arr = NULL;
+	   int ms2_peak_num = -1; double *ms2_mz_arr = NULL, *ms2_intensity_arr = NULL;
+
+	   map<int, boost::tuple<double*, double*, double*, int>>::iterator ms1_intensityIter = ms1scan_mz_intensity_rank_map->find(curr_ms1scan);
+	   if (ms1_intensityIter == ms1scan_mz_intensity_rank_map->end()) { carp(CARP_DETAILED_DEBUG, "No intensity found in MS1 scan:%d !!!", curr_ms1scan); }
+	   else {
+	       ms1_mz_arr = (ms1_intensityIter->second).get<0>();
```

Comment:
I noticed that there is a lot of ".get<0>". Would it be better to create getter functions for everything? It would be more clean than get<0>, get<1>, get<2>, etc